koog-ktor/src/commonMain/kotlin/ai/koog/ktor/utils/LLMModelParser.kt

```diff
@@ -0,0 +1,244 @@
+    ),
+    "audio" to mapOf(
+        "gpt4ominiaudio" to OpenAIModels.Audio.GPT4oMiniAudio,
+        "gpt4oaudio" to OpenAIModels.Audio.GPT4oAudio
+    ),
+    "embeddings" to mapOf(
+        "textembedding3small" to OpenAIModels.Embeddings.TextEmbedding3Small,
+        "textembedding3large" to OpenAIModels.Embeddings.TextEmbedding3Large,
+        "textembeddingada002" to OpenAIModels.Embeddings.TextEmbeddingAda002
+    ),
+    "moderation" to mapOf(
+        "text" to OpenAIModels.Moderation.Text,
+        "omni" to OpenAIModels.Moderation.Omni
+    )
+)
+
+private val ANTHROPIC_MODELS_MAP = mapOf(
+    "opus_3" to AnthropicModels.Opus_3,
+    "opus_4" to AnthropicModels.Opus_4,
+    "haiku_3" to AnthropicModels.Haiku_3,
+    "haiku_3_5" to AnthropicModels.Haiku_3_5,
+    "sonnet_3_5" to AnthropicModels.Sonnet_3_5,
+    "sonnet_3_7" to AnthropicModels.Sonnet_3_7,
+    "sonnet_4" to AnthropicModels.Sonnet_4
+)
+
+private val GOOGLE_MODELS_MAP = mapOf(
+    "gemini1_5pro" to GoogleModels.Gemini1_5Pro,
+    "gemini1_5prolatest" to GoogleModels.Gemini1_5ProLatest,
+    "gemini2_0flash" to GoogleModels.Gemini2_0Flash,
+    "gemini2_0flash001" to GoogleModels.Gemini2_0Flash001,
+    "gemini2_0flashlite" to GoogleModels.Gemini2_0FlashLite,
+    "gemini2_0flashlite001" to GoogleModels.Gemini2_0FlashLite001,
+    "gemini1_5flash" to GoogleModels.Gemini1_5Flash,
+    "gemini1_5flashlatest" to GoogleModels.Gemini1_5FlashLatest,
+    "gemini1_5flash002" to GoogleModels.Gemini1_5Flash002,
+    "gemini1_5flash8b" to GoogleModels.Gemini1_5Flash8B,
+    "gemini1_5flash8b001" to GoogleModels.Gemini1_5Flash8B001,
+    "gemini1_5flash8blatest" to GoogleModels.Gemini1_5Flash8BLatest,
+)
+
+private val OPENROUTER_MODELS_MAP = mapOf(
+    "claude3sonnet" to OpenRouterModels.Claude3Sonnet,
+    "claude3haiku" to OpenRouterModels.Claude3Haiku,
+    "gpt4" to OpenRouterModels.GPT4,
+    "gpt4o" to OpenRouterModels.GPT4o,
+    "gpt4turbo" to OpenRouterModels.GPT4Turbo,
+    "gpt35turbo" to OpenRouterModels.GPT35Turbo
+)
+
+private val OLLAMA_GROQ_MODELS_MAP = mapOf(
+    "llama3-grok-tool-use:8b" to OllamaModels.Groq.LLAMA_3_GROK_TOOL_USE_8B,
+    "llama3-groq-tool-use:8b" to OllamaModels.Groq.LLAMA_3_GROK_TOOL_USE_8B,
+    "llama3-grok-tool-use:70b" to OllamaModels.Groq.LLAMA_3_GROK_TOOL_USE_70B,
+    "llama3-groq-tool-use:70b" to OllamaModels.Groq.LLAMA_3_GROK_TOOL_USE_70B
+)
+
+private val OLLAMA_META_MODELS_MAP = mapOf(
+    "llama3.2:3b" to OllamaModels.Meta.LLAMA_3_2_3B,
+    "llama3.2" to OllamaModels.Meta.LLAMA_3_2,
+    "llama4:latest" to OllamaModels.Meta.LLAMA_4,
+    "llama-guard3:latest" to OllamaModels.Meta.LLAMA_GUARD_3
+)
+
+private val OLLAMA_ALIBABA_MODELS_MAP = mapOf(
+    "qwen2.5:0.5b" to OllamaModels.Alibaba.QWEN_2_5_05B,
+    "qwen3:0.6b" to OllamaModels.Alibaba.QWEN_3_06B,
+    "qwq:32b" to OllamaModels.Alibaba.QWQ_32B,
+    "qwq" to OllamaModels.Alibaba.QWQ,
+    "qwen2.5-coder:32b" to OllamaModels.Alibaba.QWEN_CODER_2_5_32B
+)
```

Comment:
I think we need to come up with a unified way to support models for Ollama.
Unlike other providers, it has significantly more model variations, and new ones are added more frequently.
Maintaining them in multiple places creates problems

Reply:
Totally agree, but I didn't want to impact _core_ when taking over this PR. Can we move this to a subsequent PR?

Reply:
Yeap, it’s not related to your changes. I just wanted to highlight the issue

Reply:
For Ollama, you can already do:

```kotlin
val client = OllamaClient()
val model = client.getModelOrNull(modelName, pullIfMissing = true)?.toLLModel()
```

In the future, this intermediate `OllamaModelCard` will be removed, when `LLModel` contains enough information.